heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCacheQueryUtils.java

```diff
@@ -0,0 +1,290 @@
+// Copyright 2016 Twitter. All rights reserved.
```

Comment:
I thought you were talking about the classes in the MetricsCacheQueryUtils.java

The protobuf objects MetricDatum and TmasterExceptionLog have duplicated info., such as component_name, instance_id, metric_name, which may waste space.

The MetricsDatapoint and ExceptionDatapoint are the neat data to be stored in the cache buckets. The ExceptionDatapoint may be optimized to omit the component_name and instance_id like MetricsDatapoint.